doc/book/examples/sources/patterns/hot-potato.move

```diff
@@ -1,3 +1,65 @@
 // Copyright (c) 2022, Mysten Labs, Inc.
 // SPDX-License-Identifier: Apache-2.0
 
+module examples::trade_in {
+    use sui::transfer;
+    use sui::sui::SUI;
+    use sui::coin::{Self, Coin};
+    use sui::object::{Self, Info};
+    use sui::tx_context::{TxContext};
+
+    /// Price for the first phone model in series
+    const MODEL_ONE_PRICE: u64 = 10000;
+
+    /// Price for the second phone model
+    const MODEL_TWO_PRICE: u64 = 20000;
+
+    /// For when someone tries to purchase non-existing model
+    const EWrongModel: u64 = 1;
+
+    /// For when paid amount does not match the price
+    const EIncorrectAmount: u64 = 2;
+
+    /// A phone; can be purchased or traded in for a newer model
+    struct Phone has key, store { info: Info, model: u8 }
+
+    /// Payable receipt. Has to be paid directly or paid with a trade-in option.
+    /// Cannot be stored, owned or dropped - has to be used to select one of the
+    /// options for payment: `trade_in` or `pay_full`.
+    struct Receipt { price: u64 }
```

Comment:
Should this remember the `ID` of the `Phone` it was used to purchase?

In this particular code, it's probably harmless to mix/match phones and receipts, but I can imagine a lot of cases where you would not want to allow this.

Reply:
Didn't figure out a way to check for ID as phone is never passed afterwards. Merging this in to have this example in the book.